### DISCUSSION TITLE
Get both data and count in one query

### DISCUSSION THREAD

**Discussion by riwu** [sqlalchemy/sqlalchemy#11864] [Usage Questions]
I have a query like this:
```
query = select(Table).where(...).order_by(...).limit(10)
```

To get the total count of this query result (ignoring the `limit 10`), I can add `query.with_only_columns(func.count()).order_by(None)`, which works.

However, I also want to get the actual row data.  
Is there a way to accomplish that? Something like `select([Table, func.count()])`? But count doesn't seem to work with `order_by`?

**Comment by CaselIT** [comment]
Hi,

Count is an an aggregation function, so in most sql databases you can using it alongside other non-aggregated columns.

In this case I would suggest to start by writing the sql query you want to run in the db, once you have it feel free to ask for suggestions on how to render that sql using the sqlalchemy api.

If you use postgresql (not use if other servers) you may use `func.count(1).over()` to make a window function over the complete row-set
